drivers/adc/adc_stm32.c

```diff
@@ -1174,6 +1176,56 @@ static const struct adc_driver_api api_stm32_driver_api = {
 	.ref_internal = STM32_ADC_VREF_MV, /* VREF is usually connected to VDD */
 };
 
+/*
+ * Prefix default when not configured by the DTS
+ * depending on the serie (stm32F1 has none)
+ */
+#if defined(CONFIG_SOC_SERIES_STM32L1X) || \
+	defined(CONFIG_SOC_SERIES_STM32U5X)
+#define prescaler_default	LL_ADC_CLOCK_ASYNC_DIV
+#elif !defined(CONFIG_SOC_SERIES_STM32F1X)
+#define prescaler_default	LL_ADC_CLOCK_SYNC_PCLK_DIV
+#endif
+
+/* st_prescaler property requires 2 elements : clock ASYNC/SYNC and DIV */
+#define STM32_ADC_CLOCK(x)	DT_INST_PROP_BY_IDX(x, st_adc_prescaler, 0)
+#define STM32_ADC_DIV(x)	DT_INST_PROP_BY_IDX(x, st_adc_prescaler, 1)
+
+/*
+ * Macro to check the property is valid with two elements:
+ * 1st one is SYNC or ASYNC
+ * 2nd one for the divider
+ */
+#define ADC_STM32_CHECK_PRESC(x)							\
+	IF_ENABLED(DT_INST_NODE_HAS_PROP(x, st_adc_prescaler),				\
+			(BUILD_ASSERT((DT_INST_PROP_LEN(x, st_adc_prescaler) == 2),	\
+				"Invalid ADC st_adc_prescaler property length");	\
+			BUILD_ASSERT(							\
+				(STM32_ADC_CLOCK(x) == SYNC) ||				\
+				(STM32_ADC_CLOCK(x) == ASYNC),				\
+				"Invalid ADC st_adc_prescaler property");))
+
+/* Macro to set the prefix depending on the 1st element: check if it is SYNC or ASYNC */
+#define STM32_ADC_PREFIX(x)				\
+	COND_CODE_1(IS_EQ(STM32_ADC_CLOCK(x), SYNC),	\
+		LL_ADC_CLOCK_SYNC_PCLK_DIV,		\
+		LL_ADC_CLOCK_ASYNC_DIV)
+
+/*
+ * Concat prefix (1st element) and DIV value (2nd element) of st,adc-prescaler
+ * not available for stm32F1 serie: set the config->clock_prescaler to 0 instead
+ * If the property is not defined, the default prescaler with DIV 4 is used.
+ */
+#if defined(CONFIG_SOC_SERIES_STM32F1X)
+#define DT_STM32_PRESC(x) 0
```

Comment:
The `DT_` namespace is reserved for devicetree.h. I would prefer if you renamed it to STM32_DT_PRESC.

Reply:
changed toSTM32_ADC_DT_PRESC